### PR TITLE
EL-926: Remove household feature flag and refactor

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-926)

## What changed and why

Remove the household flow feature flag and all non-household flow code
Refactor to tidy up

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: app/lib/steps/income_section.rb [hunk 1]
```diff
@@ -5,27 +5,13 @@ def all_steps
         %i[employment housing_benefit housing_benefit_details benefits benefit_details other_income outgoings]
```
Comment (willc-work) [inline app/lib/steps/income_section.rb:5]: I think we can remove `housing_benefit` `housing_benefit_details` and `outgoings` from these steps

Comment (patrick-laa) [reply app/lib/steps/income_section.rb]: Good spot! Have removed.

File: spec/forms/partner_details_form_spec.rb [hunk 2]
```diff
@@ -17,7 +16,6 @@
   end
 
   it "has some long-form copy and questions about employment" do
```
Comment (willc-work) [inline spec/forms/partner_details_form_spec.rb:18]: this test no longer checks long form content so can be renamed

**Review by willc-work — APPROVED** [review]
just a suggestion to rename a test, otherwise 👍

File: spec/forms/partner_details_form_spec.rb (hunk 2 above)
Comment (patrick-laa) [reply spec/forms/partner_details_form_spec.rb:18]: Good spot! Fixing now.

**Pull request merged**